core/services/workflows/engine.go

```diff
@@ -334,7 +334,8 @@ func (e *Engine) registerTrigger(ctx context.Context, t *triggerCapability) erro
 
 	triggerRegRequest := capabilities.CapabilityRequest{
 		Metadata: capabilities.RequestMetadata{
-			WorkflowID: e.workflow.id,
+			WorkflowID:    e.workflow.id,
+			WorkflowDonID: e.donInfo.ID,
```

Comment:
Nit: WorkflowName + WorkflowOwner should probably also be added here for consistency